### PR TITLE
Repair script for Cocoapods & Nami.xcframework

### PR DESCRIPTION
**Pull request by KiGi**
Created script meant to run in same directory as generated Pods directory, that repairs the missing Nami.xcfrmaework target inclusion in the react-native-nami-sdk library.

Script should be fairly robust, it performs several intermediate checks and also will not run if it detects changes to be made are already in place.

### CODE REVIEW

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+echo "Nami.xcframework PBXBuildFile reference is ${NAMIXC_PBXFILEREF}"
+
+
+
+# These build file reference lines will get added later in the final AWK script.
+
+# This ID is one Xcode generated for use by the Nami framework, re-using the generated code should not overlap with anything pre-existing.
+BUILD_FILE_REF="EBA7F0FF244E651F0048F322"
+BUILD_FILE_LINE=" 		${BUILD_FILE_REF} /* Nami.xcframework in Frameworks */ = {isa = PBXBuildFile; fileRef = ${NAMIXC_PBXFILEREF} /* Nami.xcframework */; };"
```
Comment (acantarero) [inline ios/scripts/fixCocoapodsProjForNamiXCFramework.sh:46]: does this require all the white space after the `"`?

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
+#!/bin/sh
+
+## Created by Kendall Gelner for NamiML on April 20th, 2020.
+## This script corrects an error in Cocopods Pods.xcodeproject generation, where Nami.xcframeowrk is included as a file reference, but is not used as a build reference for the react-native-nami-sdk bridge pulled in via Podfile.
```
Comment (acantarero) [inline ios/scripts/fixCocoapodsProjForNamiXCFramework.sh:4]: cocoapods spelled wrong on this line

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+## Run this script after "pod install", before compilation of the react natuve project.  Run in directory just above generated "Pods" directory.
```
Comment (acantarero) [inline ios/scripts/fixCocoapodsProjForNamiXCFramework.sh:6]: native spelled wrong

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh (hunk 2 above)
Comment (acantarero) [inline ios/scripts/fixCocoapodsProjForNamiXCFramework.sh:4]: framework spelled wrong in Nami.xcframeowrk

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+BUILD_FILE_LINE=" 		${BUILD_FILE_REF} /* Nami.xcframework in Frameworks */ = {isa = PBXBuildFile; fileRef = ${NAMIXC_PBXFILEREF} /* Nami.xcframework */; };"
+
+echo "Adding build file ref line"
+echo "${BUILD_FILE_LINE}"
+
+
+# Find the ID of the react-native-nami-sdk target
+TARGET_ID_REF=`grep "react-native-nami-sdk" ${PROJECT_FILE} | grep "target =" | cut -d " " -f 3`
+
+echo "Found target ID Ref for reaxt-native-nami-sdk: ${TARGET_ID_REF}"
```
Comment (acantarero) [inline ios/scripts/fixCocoapodsProjForNamiXCFramework.sh:55]: react spelled wrong

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+if [ -z "${TARGET_ID_REF}" ]
+then
+    echo "Could not find target ID for react-native-nami-sdk, make sure bridge is added to pacakges.json"
```
Comment (acantarero) [inline ios/scripts/fixCocoapodsProjForNamiXCFramework.sh:59]: packages spelled wrong

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    exit
+fi
+
+
+# Now find the ID for the incldued frameworks section of the react-native-nami-sdk
```
Comment (acantarero) [inline ios/scripts/fixCocoapodsProjForNamiXCFramework.sh:64]: included spelled wrong

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+echo "Found react-native-nami-sdk Framework Build Phase ID  ${FRAMEWORK_BUILD_PHASE_ID}"
+
+if [ -z "${FRAMEWORK_BUILD_PHASE_ID}" ]
+then
+    echo "Could not find Framework section reference ID for react-native-nami-sdk, make sure bridge is added to pacakges.json"
+    exit
+fi
+
+
+FRAMEWORK_BUILD_PHASE_LINE="				${BUILD_FILE_REF} /* Nami.xcframework in Frameworks */,"
```
Comment (acantarero) [inline ios/scripts/fixCocoapodsProjForNamiXCFramework.sh:85]: does this line require so much white space after first `"`?

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh (hunk 1 above)
Comment (KiGi) [reply ios/scripts/fixCocoapodsProjForNamiXCFramework.sh:46]: Yep!  It's meant to be as close as possible to what Xcode would insert.

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh (hunk 7 above)
Comment (KiGi) [reply ios/scripts/fixCocoapodsProjForNamiXCFramework.sh:85]: Also the same thing as what Xcode would put into the file, best to leave it just as Xcode would insert it.

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh (hunk 4 above)
Comment (KiGi) [reply ios/scripts/fixCocoapodsProjForNamiXCFramework.sh]: Fixed.

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh (hunk 6 above)
Comment (KiGi) [reply ios/scripts/fixCocoapodsProjForNamiXCFramework.sh]: Fixed.

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh (hunk 5 above)
Comment (KiGi) [reply ios/scripts/fixCocoapodsProjForNamiXCFramework.sh]: Fixed.

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh (hunk 2 above)
Comment (KiGi) [reply ios/scripts/fixCocoapodsProjForNamiXCFramework.sh]: Fixed.

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh (hunk 3 above)
Comment (KiGi) [reply ios/scripts/fixCocoapodsProjForNamiXCFramework.sh]: Fixed.

File: ios/scripts/fixCocoapodsProjForNamiXCFramework.sh (hunk 2 above)
Comment (KiGi) [reply ios/scripts/fixCocoapodsProjForNamiXCFramework.sh]: Fixed.

**Comment by acantarero** [comment]
was merged into other PR

**Pull request closed without merging**